### PR TITLE
13897-make-a-copy-collections

### PR DESCRIPTION
**Pull request by kowalczykpawel7**
Feature #13897

Arvados-DCO-1.1-Signed-off-by: Pawel Kowalczyk <pawel.kowalczyk@contractors.roche.com>

### CODE REVIEW

File: src/store/collection-copy-dialog/collection-copy-dialog.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import { resetPickerProjectTree } from '~/store/project-tree-picker/project-tree-picker-actions';
+import { RootState } from '~/store/store';
+import { ServiceRepository } from '~/services/services';
+import { getCommonResourceServiceError, CommonResourceServiceError } from '~/common/api/common-resource-service';
+import { snackbarActions } from '../snackbar/snackbar-actions';
+import { projectPanelActions } from '~/store/project-panel/project-panel-action';
+
+export const COLLECTION_COPY_DIALOG = 'projectCopy';
+
+export interface CopyFormDialogData {
```
Comment (burzomir) [inline src/store/collection-copy-dialog/collection-copy-dialog.ts]: I think that interface name should contain `Collection`.

File: src/store/collections/collection-copy-actions.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+import { dialogActions } from "~/store/dialog/dialog-actions";
+import { initialize, startSubmit, stopSubmit } from 'redux-form';
+import { resetPickerProjectTree } from '~/store/project-tree-picker/project-tree-picker-actions';
+import { RootState } from '~/store/store';
+import { ServiceRepository } from '~/services/services';
+import { getCommonResourceServiceError, CommonResourceServiceError } from '~/common/api/common-resource-service';
+import { snackbarActions } from '~/store/snackbar/snackbar-actions';
+import { projectPanelActions } from '~/store/project-panel/project-panel-action';
+
+export const COLLECTION_COPY_DIALOG = 'projectCopy';
```
Comment (janicka1) [inline src/store/collections/collection-copy-actions.ts]: To be consist with whole app I recommend to change name for: `COLLECTION_COPY_FORM_NAME`

File: src/store/collections/collection-copy-actions.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+export interface CollectionCopyFormDialogData {
+    name: string;
+    ownerUuid: string;
+    uuid: string;
+}
+
+export const openCollectionCopyDialog = (resource: { name: string, uuid: string }) =>
```
Comment (janicka1) [inline src/store/collections/collection-copy-actions.ts]: +1 for naming

File: src/views-components/dialog-collection-copy/dialog-collection-copy.tsx [hunk 4]
```diff
@@ -7,28 +7,28 @@ import { InjectedFormProps, Field } from 'redux-form';
...
+import { COPY_NAME_VALIDATION, COPY_FILE_VALIDATION } from '~/validators/validators';
 import { TextField } from "~/components/text-field/text-field";
-import { ProjectCopyFormDialogData } from "~/store/project-copy-dialog/project-copy-dialog";
+import { CollectionCopyFormDialogData } from "~/store/collections/collection-copy-actions";
 
-type ProjectCopyFormDialogProps = WithDialogProps<string> & InjectedFormProps<ProjectCopyFormDialogData>;
+type CopyFormDialogProps = WithDialogProps<string> & InjectedFormProps<CollectionCopyFormDialogData>;
 
-export const ProjectCopyFormDialog = (props: ProjectCopyFormDialogProps) =>
+export const DialogCopy = (props: CopyFormDialogProps) =>
```
Comment (janicka1) [inline src/views-components/dialog-collection-copy/dialog-collection-copy.tsx]: I recommend change this name for `DialogCollectionCopy`

File: src/views-components/dialog-collection-copy/dialog-collection-copy.tsx [hunk 5]
```diff
@@ -7,28 +7,28 @@ import { InjectedFormProps, Field } from 'redux-form';
...
     <FormDialog
         dialogTitle='Make a copy'
-        formFields={ProjectCopyFields}
+        formFields={DialogCopyFields}
```
Comment (janicka1) [inline src/views-components/dialog-collection-copy/dialog-collection-copy.tsx]: I recommend change this name for `CollectionCopyFields`

File: src/store/collections/collection-copy-actions.ts [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+import { dialogActions } from "~/store/dialog/dialog-actions";
+import { initialize, startSubmit, stopSubmit } from 'redux-form';
+import { resetPickerProjectTree } from '~/store/project-tree-picker/project-tree-picker-actions';
+import { RootState } from '~/store/store';
+import { ServiceRepository } from '~/services/services';
+import { getCommonResourceServiceError, CommonResourceServiceError } from '~/common/api/common-resource-service';
+import { snackbarActions } from '~/store/snackbar/snackbar-actions';
+import { projectPanelActions } from '~/store/project-panel/project-panel-action';
+
+export const COLLECTION_COPY_FORM_NAME = 'projectCopy';
```
Comment (janicka1) [inline src/store/collections/collection-copy-actions.ts]: 'CollectionCopyDialog'

File: src/views-components/dialog-forms/copy-collection-dialog.ts [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+import { reduxForm } from 'redux-form';
+import { COLLECTION_COPY_FORM_NAME, CollectionCopyFormDialogData, copyCollection } from '~/store/collections/collection-copy-actions';
+import { DialogCollectionCopy } from "~/views-components/dialog-collection-copy/dialog-collection-copy";
+
+export const CopyCollectionDialog = compose(
+    withDialog(COLLECTION_COPY_FORM_NAME),
+    reduxForm<CollectionCopyFormDialogData>({
+        form: COLLECTION_COPY_FORM_NAME,
+        onSubmit: (data, dispatch) => {
+            dispatch(copyCollection(data));
```
Comment (janicka1) [inline src/views-components/dialog-forms/copy-collection-dialog.ts]: `DialogCollectionCopy `

**Pull request merged**